File: java/src/Board.java
import java.util.ArrayList;

public class Board {

    ArrayList<Square> squares;

    public Board(){
        squares = new ArrayList<Square>();
        for(int i=0;i<3;i++){
            for(int j=0;j<3;j++){
                squares.add(new Square(i,j));
            }
        }
    }

    public ArrayList<Square> getSquares() {
        return squares;
    }
}

File: src/Board.js
import React from 'react';
import Square from './Square';

function Board(props) {
    function renderSquare(i) {
        return (
            <Square
                value={props.squares[i]}
                onClick={() => props.onClick(i)}
                winner={props.winner}
                index={i}
            />
        );
    }

    function renderRow(j) {
        let row = [];
        for(let i = j; i <= j + 2; i++) {
            row.push(renderSquare(i));
        }

        return row;
    }

    function renderBoard(k) {
        let board = [];
        for(let m = 0; m < k; m += 3) {
            board.push(<div className="board-row">{renderRow(m)}</div>);
        }
        return board;
    }

    return (
        <div>
            <div className="better-board">
                {renderBoard(9)}
            </div>
        </div>
    );
}

export default Board;
